buildbot/tox_setup.sh

```diff
@@ -1,74 +1,102 @@
-    cd ${PYENV_ROOT}/plugins/pyenv-virtualenv
-    git checkout `git tag | tail -1`
+if [[ ! -d $PROJDIR/riak_pb/src ]]
+then
+    (cd $PROJDIR && git submodule update --init)
 fi
 
-if [[ ! -d ${PYENV_ROOT}/plugins/pyenv-alias ]]; then
-    git clone https://github.com/s1341/pyenv-alias.git ${PYENV_ROOT}/plugins/pyenv-alias
+# Install pyenv if it's missing
+if [[ ! -d $PYENV_ROOT ]]
+then
+    git clone https://github.com/yyuu/pyenv.git $PYENV_ROOT
+else
+    (cd $PYENV_ROOT && git fetch --all)
 fi
 
-# Add pyenv root to PATH
-# and initialize pyenv
-PATH="$PYENV_ROOT/bin:$PATH"
-# initialize pyenv
-eval "$(pyenv init -)"
-# initialize pyenv virtualenv
-eval "$(pyenv virtualenv-init -)"
+(cd $PYENV_ROOT && git checkout $(git describe --tags $(git rev-list --tags --max-count=1)))
 
-# Now install (allthethings) versions for testing
-if [[ -z $(pyenv versions | grep riak_3.4.3) ]]; then
-    VERSION_ALIAS="riak_3.4.3" pyenv install 3.4.3
-    pyenv virtualenv riak_3.4.3 riak-py34
+declare -r pyenv_virtualenv_dir="$PYENV_ROOT/plugins/pyenv-virtualenv"
+if [[ ! -d $pyenv_virtualenv_dir ]]
+then
+    git clone https://github.com/yyuu/pyenv-virtualenv.git $pyenv_virtualenv_dir
+else
+    (cd $pyenv_virtualenv_dir && git fetch --all)
 fi
-if [[ -z $(pyenv versions | grep riak_3.3.6) ]]; then
-    VERSION_ALIAS="riak_3.3.6" pyenv install 3.3.6
-    pyenv virtualenv riak_3.3.6 riak-py33
-fi
-if [[ -z $(pyenv versions | grep riak_2.7.10) ]]; then
-    VERSION_ALIAS="riak_2.7.10" pyenv install 2.7.10
-    pyenv virtualenv riak_2.7.10 riak-py27
+
+(cd $pyenv_virtualenv_dir && git checkout $(git describe --tags $(git rev-list --tags --max-count=1)))
+
+declare -r pyenv_alias_dir="$PYENV_ROOT/plugins/pyenv-alias"
+if [[ ! -d $pyenv_alias_dir ]]
+then
+    git clone https://github.com/s1341/pyenv-alias.git $pyenv_alias_dir
+else
+    (cd $pyenv_alias_dir && git pull origin master)
 fi
-if [[ -z $(pyenv versions | grep riak_2.7.9) ]]; then
-    VERSION_ALIAS="riak_2.7.9" pyenv install 2.7.9
-    pyenv virtualenv riak_2.7.9 riak-py279
+
+# Add pyenv root to PATH
+# and initialize pyenv
+if [[ $PATH != */.pyenv* ]]
+then
+    echo "[INFO] adding $PYENV_ROOT/bin to PATH"
+    export PATH="$PYENV_ROOT/bin:$PATH"
 fi
-if [[ -z $(pyenv versions | grep riak_2.6.9) ]]; then
-    VERSION_ALIAS="riak_2.6.9" pyenv install 2.6.9
-    pyenv virtualenv riak_2.6.9 riak-py26
+
+if [[ $(type -t pyenv) != 'function' ]]
+then
+    echo "[INFO] init pyenv"
+    eval "$(pyenv init -)"
+    eval "$(pyenv virtualenv-init -)"
 fi
-pyenv global riak-py34 riak-py33 riak-py27 riak-py279 riak-py26
-pyenv versions
 
-# Now install tox
-pip install --upgrade pip
-if [ -z "`pip show tox`" ]; then
-    pip install -Iv tox
-    if [ -z "`pip show tox`" ]; then
-        echo "ERROR: Install of tox failed"
-        exit 1
+# Now install (allthethings) versions for testing
+for pyver in 2.7 3.3 3.4 3.5
```

Comment:
I *know* that relatively current distros ship with old Pythons as their defaults. Ubuntu 14.04 ships with 2.7.3 and Centos 7 ships with 2.7.5, for example.